app/products/calls/components/current_call_bar/current_call_bar.tsx

```diff
@@ -0,0 +1,184 @@
+            borderRadius: 5,
+            padding: 4,
+            height: CURRENT_CALL_BAR_HEIGHT - 10,
+            alignItems: 'center',
+        },
+        pressable: {
+            zIndex: 10,
+        },
+        userInfo: {
+            flex: 1,
+        },
+        speakingUser: {
+            color: theme.sidebarText,
+            fontWeight: '600',
+            fontSize: 16,
+        },
+        currentChannel: {
+            color: theme.sidebarText,
+            opacity: 0.64,
+        },
+        micIcon: {
+            color: theme.sidebarText,
+            width: 42,
+            height: 42,
+            textAlign: 'center',
+            textAlignVertical: 'center',
+            justifyContent: 'center',
+            backgroundColor: '#3DB887',
+            borderRadius: 4,
+            margin: 4,
+            padding: 9,
+            overflow: 'hidden',
+        },
+        muted: {
+            backgroundColor: 'transparent',
+        },
+        expandIcon: {
+            color: theme.sidebarText,
+            padding: 8,
+            marginRight: 8,
+        },
+    };
+});
+
+const CurrentCallBar = ({
+    displayName,
+    currentCall,
+    userModelsDict,
+    teammateNameDisplay,
+}: Props) => {
+    const theme = useTheme();
+    const isCurrentCall = Boolean(currentCall);
+    const [speaker, setSpeaker] = useState<string | null>(null);
+    const handleVoiceOn = (data: VoiceEventData) => {
+        if (data.channelId === currentCall?.channelId) {
+            setSpeaker(data.userId);
+        }
+    };
+    const handleVoiceOff = (data: VoiceEventData) => {
+        if (data.channelId === currentCall?.channelId && ((speaker === data.userId) || !speaker)) {
+            setSpeaker(null);
+        }
+    };
+
+    useEffect(() => {
+        const onVoiceOn = DeviceEventEmitter.addListener(WebsocketEvents.CALLS_USER_VOICE_ON, handleVoiceOn);
+        const onVoiceOff = DeviceEventEmitter.addListener(WebsocketEvents.CALLS_USER_VOICE_OFF, handleVoiceOff);
+        DeviceEventEmitter.emit(Events.CURRENT_CALL_BAR_VISIBLE, isCurrentCall);
+        return () => {
+            DeviceEventEmitter.emit(Events.CURRENT_CALL_BAR_VISIBLE, Boolean(false));
+            onVoiceOn.remove();
+            onVoiceOff.remove();
+        };
+    }, [isCurrentCall]);
+
+    const goToCallScreen = useCallback(() => {
+        const options: Options = {
+            layout: {
+                backgroundColor: '#000',
+                componentBackgroundColor: '#000',
+                orientation: ['portrait', 'landscape'],
+            },
+            topBar: {
+                background: {
+                    color: '#000',
+                },
+                visible: Platform.OS === 'android',
+            },
+        };
+        goToScreen('Call', 'Call', {}, options);
+    }, []);
+
+    const myParticipant = currentCall?.participants[currentCall.myUserId];
+    if (!currentCall || !myParticipant) {
+        return null;
+    }
```

Comment:
we should be preventing this component from mounting if any of this two aren't true.

Reply:
Thanks -- it isn't being mounted now. (this was mostly so we didn't have to do the `?.serverUrl || ''` thing below).  Removed unneeded null: 9af9912de3b7842a679c6406fefdf0edeb944faf